### PR TITLE
Ddb connector with extractor and GeneratedRowWriter

### PR DESCRIPTION
**Pull request by nauy2697**
*Issue #, if available:*
N/A

*Description of changes:*
Improve DynamoDB connector with Extractor framework. 
Current data types supported for extractor framework. 

1. Bit
2. Binary
3. Decimal
4. Varchar

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by nauy2697** [comment]
> did you run integration tests?

Integration tests performed with `DynamoDbIntegTest` 
Test passed

File: athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBRecordHandler.java [hunk 1]
```diff
@@ -138,66 +139,45 @@ protected void readWithConstraint(BlockSpiller spiller, ReadRecordsRequest recor
...
+                // this can happen regardless of the hasNext() check above for the very first iteration since itemIterator
+                // had not made any DDB calls yet and there may be zero items returned when it does
+                continue;
+            }
+            numRows++;
+            try {
+                spiller.writeRows((Block block, int rowNum) -> rowWriter.writeRow(block, rowNum, item) ? 1 : 0);
+            }
+            catch (Exception ex) {
+                logger.warn("Exception while writing rows :{}", ex.getMessage(), ex);
```
Comment (henrymai) [inline athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/DynamoDBRecordHandler.java:177]: Is it safe to continue to write the other rows when this exception happens?

Also what about `numRows`, it was incremented before this exception happened, should `numRows` be incremented inside the `try{}` after `writeRows` succeeds instead?

File: athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/util/DDBTypeUtils.java [hunk 2]
```diff
@@ -347,4 +365,116 @@ public static List<Object> coerceListToExpectedType(Object value, Field field, D
...
+
+                            return true;
+                        };
+            case STRUCT:
+                return (FieldVector vector, Extractor extractor, ConstraintProjector constraint) ->
+                        (FieldWriter) (Object context, int rowNum) ->
+                        {
+                            Map<String, AttributeValue> item = (Map<String, AttributeValue>) context;
+                            Object value = ItemUtils.toSimpleValue(item.get(field.getName()));
+                            value = DDBTypeUtils.coerceValueToExpectedType(value, field, fieldType, recordMetadata);
```
Comment (henrymai) [inline athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/util/DDBTypeUtils.java:464]: Just wondering, do we have to worry about `value` being `null` here?

File: athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/util/DDBTypeUtils.java [hunk 3]
```diff
@@ -347,4 +365,116 @@ public static List<Object> coerceListToExpectedType(Object value, Field field, D
...
+                            BlockUtils.setComplexValue(vector, rowNum, resolver, value);
+                            return true;
+                        };
+            default:
+                return (FieldVector vector, Extractor extractor, ConstraintProjector constraint) ->
+                        (FieldWriter) (Object context, int rowNum) ->
+                        {
+                            Map<String, AttributeValue> item = (Map<String, AttributeValue>) context;
+                            Object value = ItemUtils.toSimpleValue(item.get(field.getName()));
+                            value = DDBTypeUtils.coerceValueToExpectedType(value, field, fieldType, recordMetadata);
```
Comment (henrymai) [inline athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/util/DDBTypeUtils.java:474]: Same question as above.

**Pull request closed without merging**